src/extension/provider/statefulAuth.ts

```diff
@@ -248,6 +237,71 @@ export class StatefulAuthProvider implements AuthenticationProvider, Disposable
     this.#disposables.forEach((d) => d.dispose())
   }
 
+  public async loadFromSecrets() {
+    const secretsFile = '/etc/secrets/playground-dev'
+    const secretsUri = Uri.parse(secretsFile)
+    const hasSecretsFile = await workspace.fs.stat(secretsUri).then(
+      () => true,
+      () => false,
+    )
+
+    if (!hasSecretsFile) {
+      logger.info('No secrets file found, skipping load from secrets')
+      return
+    }
+
+    const rawToken = await workspace.fs.readFile(secretsUri)
+    if (!rawToken?.length) {
+      logger.error('Failed to read secrets file')
+      return
+    }
+
+    const token = new TextDecoder().decode(rawToken).trim()
```

Comment:
259-270 should go into a protected/private method `InsecureDecode` that's merely being called here. The code does not indicate by itself that the JWT is *not* being verified until the API is called further down.